zebra/dplane_fpm_nl.c

```diff
@@ -1680,6 +1680,25 @@ static int fpm_nl_process(struct zebra_dplane_provider *prov)
 
 	fnc = dplane_provider_get_data(prov);
 	limit = dplane_provider_get_work_limit(prov);
+
+	frr_with_mutex (&fnc->ctxqueue_mutex) {
+		cur_queue = dplane_ctx_queue_count(&fnc->ctxqueue);
+	}
+
+	if (cur_queue >= (uint64_t)limit) {
+		if (IS_ZEBRA_DEBUG_FPM)
+			zlog_debug("%s: Already at a limit(%" PRIu64
+				   ") of internal work, hold off",
+				   __func__, cur_queue);
+		limit = 0;
+	} else {
+		if (IS_ZEBRA_DEBUG_FPM)
+			zlog_debug("%s: current queue is %" PRIu64
```

Comment:
maybe this should just be when there is queue, like  `if (cur_queue > 0) ...`

Reply:
I'm a bit confused I have two situations I want to display debugs for.  a) When the limit is already reached for the particular input queue so we are not moving anything or b) How many items we are moving into the input queue.

What would you have me do differently?

Reply:
I guess there are three cases:
1. no queue
2. queue, but under the limit
3. queue at or over the limit
the "limiting to lesser" debug will go off for 1 and 2 - but I think there's no need for a debug in case 1?

Reply:
fixed